### PR TITLE
Configure webpack for filling bundle file

### PR DESCRIPTION
**Pull request by Sasha1152**
http://ssu-jira.softserveinc.com/browse/PYTHON-98

## Code reviewers

- [ ] @meelros
- [ ] @BoroviyOrest
- [ ] @Kardan1123
- [ ] @mizin4ik
- [ ] @dimigor
- [ ] @PetrushynskyiOleksii

### Second Level Review

- [ ] @lhalam

## Summary of change

Created directories: 'way_to_home/static/src', 'way_to_home/static/public'
Created files: 'webpack.config.js', 'package.json' both in the root directory,
the file 'bundle.js' in the directory 'way_to_home/static/public',
the file 'app.js' in the directory 'way_to_home/static/src',
the file 'index.js' in the directory 'way_to_home/static'.

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+
+module.exports = {
+
+    mode: 'development',
```
Comment (lhalam) [inline webpack.config.js]: is it necessary?

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    entry: './way_to_home/static/index.js',
```
Comment (lhalam) [inline webpack.config.js]: use path.join
The entry point path should be `way_to_home/static/src/app.js`

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    output: {
+        path: path.resolve(__dirname, 'dist'),
```
Comment (lhalam) [inline webpack.config.js]: use path `way_to_home/static/public`

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        filename: 'bundle.js',
+    },
+
+    watch: true,
```
Comment (lhalam) [inline webpack.config.js]: is it necessary?

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    watchOptions: {
```
Comment (lhalam) [inline webpack.config.js]: is it necessary?

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        ]
+    },
+
+    devtool: "source-map",
+
+    externals: {
+        react: 'react'
+    },
+
+    devServer: {
```
Comment (lhalam) [inline webpack.config.js]: it is not needed!

File: webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    module: {
+        rules: [
+              { test: /\.css$/, use: 'css-loader' },
+              { test: /\.ts$/, use: 'ts-loader' }
+        ]
+    },
+
+    devtool: "source-map",
+
+    externals: {
```
Comment (lhalam) [inline webpack.config.js]: it is not needed!

File: webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    },
+
+    module: {
+        rules: [
+              { test: /\.css$/, use: 'css-loader' },
+              { test: /\.ts$/, use: 'ts-loader' }
+        ]
+    },
+
+    devtool: "source-map",
```
Comment (lhalam) [inline webpack.config.js]: it is not needed!

File: webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    externals: {
+        react: 'react'
+    },
+
+    devServer: {
+        proxy: {
+            '/api': 'http://localhost:8000'
+        },
+    }
+};
```
Comment (lhalam) [inline webpack.config.js]: add endline

File: webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    watch: true,
+
+    watchOptions: {
+        aggregateTime: 100
+    },
+
+    module: {
+        rules: [
+              { test: /\.css$/, use: 'css-loader' },
+              { test: /\.ts$/, use: 'ts-loader' }
```
Comment (lhalam) [inline webpack.config.js]: it is not necessary! because we will not use TypeScript!

File: webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        filename: 'bundle.js',
+    },
+
+    watch: true,
+
+    watchOptions: {
+        aggregateTime: 100
+    },
+
+    module: {
```
Comment (lhalam) [inline webpack.config.js:14]: loaders for js/jsx

**Review by lhalam — COMMENTED** [review]
create `app.js` file
create `package.json`

`index.js` is the traditional and actual entry point for all node apps. Here in react it just has code of what to render and where to render.

`App.js` on the other hand has the root component of the react app because every view and component are handled with hierarchy in React, where `<App />` is the top most component in hierarchy. This gives you feel that you maintain hierarchy in your code starting from `App.js`.

Other than that, you can have the App logic in `index.js` file itself. But it's something related to conventions followed by the community using the library or framework. It's always feel good to go along the community.

File: webpack.config.js (hunk 1 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 2 above)
Comment (Sasha1152) [reply webpack.config.js]: fixed

File: webpack.config.js (hunk 3 above)
Comment (Sasha1152) [reply webpack.config.js]: fixed

File: webpack.config.js (hunk 4 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 5 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 7 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 8 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 9 above)
Comment (Sasha1152) [reply webpack.config.js]: done

File: webpack.config.js (hunk 10 above)
Comment (Sasha1152) [reply webpack.config.js]: deleted

File: webpack.config.js (hunk 11 above)
Comment (Sasha1152) [reply webpack.config.js:14]: added

**Review by lhalam — CHANGES_REQUESTED** [review]
ToDo
Sasha1152: fixed

File: package.json [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+  "version": "0.1.0",
+  "description": "The main idea of project – planning routes and get a notification",
+  "main": "app.js",
+  "scripts": {
+    "start": "webpack -d --watch --config webpack.config.js",
+    "build": "webpack --config webpack.config.js",
+    "test": "nyc --reporter=lcov --reporter=text webpack --config webpack-test.config.js",
+    "lint": "eslint --config .eslintrc.js eventually/static/src/;exit 0"
+  }
+};
```
Comment (lhalam) [inline package.json]: add endline

File: way_to_home/.pylintrc [hunk 13]
```diff
@@ -7,7 +7,7 @@ extension-pkg-whitelist=
 
 # Add files or directories to the blacklist. They should be base names, not
 # paths.
-ignore=CVS, migrations, apps.py, settings.py, local_settings.py, manage.py, unittests, urls.py
+ignore=CVS, migrations, apps.py, settings.py, local_settings.py, manage.py, unittests, urls.py, bundle.js, static
```
Comment (lhalam) [inline way_to_home/.pylintrc]: remove bundle.js because of it contained in the static folder

**Review by lhalam — CHANGES_REQUESTED** [review]
remove way_to_home/static/index.js file
remove way_to_home/static/public/bundle.js file  because of it contained in the .gitignore file
update app.js file? for example:
import React from 'react';
```
import React from 'react';
import ReactDOM from 'react-dom';

ReactDOM.render(
    document.getElementById('app')
);
```

File: package.json [hunk 14]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "wayToHome",
+  "version": "0.1.0",
+  "description": "The main idea of project – planning routes and get a notification",
+  "main": "app.js",
+  "scripts": {
+    "start": "webpack -d --watch --config webpack.config.js",
+    "build": "webpack --config webpack.config.js",
+    "test": "nyc --reporter=lcov --reporter=text webpack --config webpack-test.config.js",
+    "lint": "eslint --config .eslintrc.js eventually/static/src/;exit 0"
```
Comment (PetrushynskyiOleksii) [inline package.json]:
```suggestion
    "lint": "eslint --config .eslintrc.js eventually/static/src/;exit 0"
```
Change path of eslintr for our project, not for a `eventually`.

File: package.json (hunk 12 above)
Comment (Sasha1152) [reply package.json]: done

File: way_to_home/.pylintrc (hunk 13 above)
Comment (Sasha1152) [reply way_to_home/.pylintrc]: deleted

File: package.json (hunk 14 above)
Comment (Sasha1152) [reply package.json]: fixed

**Comment by Sasha1152** [comment]
This branch was remade. Look at Feature/webpack_setup branch

**Pull request closed without merging**